design/sketch.webidl

```diff
@@ -428,14 +428,43 @@ interface WebGPURenderPipeline {
 // COMMAND RECORDING (Command buffer and all relevant structures)
 // ****************************************************************************
 
-enum WebGPULoadOp {
+/// Common interface for render and compute pass encoders.
+interface WebGPUProgrammablePassEncoder {
+    WebGPUCommandBuffer end_pass();
+    // Allowed in both compute and render passes
+    //TODO: setPushConstants() ?
+    void setBindGroup(u32 index, WebGPUBindGroup bindGroup);
+    void setPipeline((WebGPUComputePipeline or WebGPURenderPipeline) pipeline);
+};
+
+interface WebGPURenderPassEncoder: WebGPUProgrammablePassEncoder {
+    void setBlendColor(float r, float g, float b, float a);
+    void setIndexBuffer(WebGPUBuffer buffer, u32 offset);
+    void setVertexBuffers(u32 startSlot, sequence<WebGPUBuffer> buffers, sequence<u32> offsets);
+
+    void draw(u32 vertexCount, u32 instanceCount, u32 firstVertex, u32 firstInstance);
+    void drawIndexed(u32 indexCount, u32 instanceCount, u32 firstIndex, u32 firstInstance, u32 firstVertex);
+
+    // TODO add missing commands
+};
+
+interface WebGPUComputePassEncoder: WebGPUProgrammablePassEncoder {
+    void dispatch(u32 x, u32 y, u32 z);
+
+    // TODO add missing commands
+};
+
+
+typedef u32 WebGPULoadOpEnum;
+interface WebGPULoadOp {
     "clear",
     "load",
 };
 
-enum WebGPUStoreOp {
+typedef u32 WebGPUStoreOpEnum;
+interface WebGPUStoreOp {
     "store",
-}
+};
```

Comment:
I'm probably missing something, but why have WebGPULoadOp and WebGPUStoreOp become interfaces instead of enums? This doesn't seem quite right (WebGPUStoreOpEnum seems unused, is "store" supposed to be a WebGPUStoreOpEnum?)